components/codec/src/buffer.rs

```diff
@@ -421,15 +444,30 @@ mod tests {
                 slice[..payload_len].clone_from_slice(payload.as_slice());
             }
 
+            // These are trying to defeat optimizations in malloc that might
+            // cause realloc to not actually create a new allocation. See
+            // the FIXME above.
+            let _alloc_noise: Vec<u8> = Vec::with_capacity(payload_len);
+            let _alloc_noise: Vec<u8> = Vec::with_capacity(1);
+
             // Re-allocate the vector space and ensure that the address is changed.
             vec.reserve(::std::cmp::max(payload_len * 3, 32));
-            assert_ne!(vec_ptr, vec.as_ptr());
+
+            //assert_ne!(vec_ptr, vec.as_ptr());
+            if vec_ptr == vec.as_ptr() {
+                in_place_reallocs += 1;
+                println!("didn't realloc at payload_len: {}", payload_len);
```

Comment:
maybe better to remove println?

Reply:
How about failing the test when none of the reallocation allocate at a different address? (but we allow some of the reallocation take in place)

Reply:
FWIW the printlns here normally are only printed when the test fails, because the test runner captures stdio, but I'm going to remove the printlns because probably nobody will ever need them.